cli/util.go

```diff
@@ -15,6 +18,13 @@ import (
 	"github.com/filecoin-project/lotus/chain/types"
 )
 
+// Set the global default, to be overridden by individual cli flags in order
+func init() {
+	color.NoColor = os.Getenv("GOLOG_LOG_FMT") != "color" &&
+		!isatty.IsTerminal(os.Stdout.Fd()) &&
```

Comment:
Strictly I think you should check `GOLOG_OUTPUT` to see if logging is being redirected to `stderr` here

Reply:
Urgh... that's a bit of a can of worms...
Perhaps we should just make this a public method and inspect the result?
https://github.com/ipfs/go-log/blob/5b4daa4491c2cc9e0ba85ce4b5fcc913f6fcdf6b/setup.go#L274-L377

Reply:
Sensible suggestion to avoid duplicating that logic

Reply:
Need to think about this more, that'll have to come via another PR